sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -2575,13 +2580,13 @@ class Analyzer(
         case ne: NamedExpression =>
           // If a named expression is not in regularExpressions, add it to
           // extractedExprBuffer and replace it with an AttributeReference.
+          val attr = ne.toAttribute
```

Comment:
is it really safe? it looks risky that we silently drop the `Alias` and assume the `Alias` expr ID has a match in the input attributes.

Reply:
hm, looking around the related code, I thought `Alias` is stored in `extractedExprBuffer` then it is always evaluated in the child logical plan (`Project` or `Aggregate`) of a `Window` node. Is there still the case where `Alias` is silently dropped?